CMakeLists.txt

```diff
@@ -414,12 +414,8 @@ if(UMF_BUILD_BENCHMARKS)
     add_subdirectory(benchmark)
 endif()
 
-if(UMF_BUILD_EXAMPLES)
-    if(NOT UMF_DISABLE_HWLOC)
-        add_subdirectory(examples)
-    else()
-        message(WARNING "Examples cannot be build - hwloc disabled")
-    endif()
+if(UMF_BUILD_EXAMPLES AND NOT UMF_DISABLE_HWLOC)
+    add_subdirectory(examples)
```

Comment:
I think you could leave a STATUS message indicating that no examples will be built. This way, if someone sets UMF_BUILD_EXAMPLES and disables hwloc at the same time, they will understand why the examples weren't built.